tests/integration_tests/instances.py

```diff
@@ -193,9 +194,20 @@ def upgrade_cloud_init(self):
         assert self.execute("apt-get update -q").ok
         assert self.execute("apt-get install -qy cloud-init").ok
 
+    def ip(self) -> str:
+        if self._ip:
+            return self._ip
+        try:
+            self._ip = self.instance.ip
+            return self._ip
+        except NotImplementedError:
+            return "Unknown"
```

Comment:
Nit: Should this be `self._ip = "Unknown"` and then return after the try...except block?